hal/inc/core_hal.h

```diff
@@ -136,12 +136,18 @@ int HAL_Core_Get_Last_Reset_Info(int *reason, uint32_t *data, void *reserved);
  */
 void HAL_Notify_Button_State(uint8_t button, uint8_t state);
 
+typedef enum {
+    HAL_STANDBY_MODE_FLAG_NONE = 0,
+    HAL_STANDBY_MODE_FLAG_DISABLE_WKP_PIN = 1
+} hal_standby_mode_flag_t;
+
 void HAL_Core_Enter_Safe_Mode(void* reserved);
 void HAL_Core_Enter_Bootloader(bool persist);
 void HAL_Core_Enter_Stop_Mode(uint16_t wakeUpPin, uint16_t edgeTriggerMode, long seconds);
 void HAL_Core_Execute_Stop_Mode(void);
-void HAL_Core_Enter_Standby_Mode(uint32_t seconds, void* reserved);
+void HAL_Core_Enter_Standby_Mode(uint32_t seconds, uint32_t flags);
```

Comment:
can we make the 2nd param a struct so that we have some extensibility in future?